### PR TITLE
[1.x] Delete the users profile picture

### LINKED ISSUE #396: Deleting a profile does not delete avatar
- Jetstream Version: "1.5.1"
- Jetstream Stack: Inertia 
- Laravel Version: 8.11.2
- PHP Version: 7.4.10
- Database Driver & Version: MySQL & don't know rn - not necessary I guess?

### Description:
When deleting your account in Jetstream, the associated image it not deleted. 

### Steps To Reproduce:
- install a fresh laravel 8 project using --jet and teams
- create an account
- delete the account in the profile section
- check disk, image is still given

### Possible Fix:
I'd assume calling unlink/delete on the file using the path from the users table `profile_photo_path` field should fix it. https://github.com/laravel/jetstream/blob/6d7030031c20249bd9a0795819918f986cebe81f/stubs/app/Actions/Jetstream/DeleteUser.php#L15

### PR DESCRIPTION
**Pull request by lostdesign**
When a user deletes an account, the image will now be deleted as well. fixes #396

### CODE REVIEW

**Comment by lostdesign** [comment]
I just realized that the profile picture feature has to be enabled, otherwise the method to delete the profile image is not available on the User model. Guess this should then be rather added to the docs? 

So when a user is deleted and the profile feature is enabled, it should also delete the image or more like they need to call `$user->deleteProfilePhoto()`

File: stubs/app/Actions/Jetstream/DeleteUserWithTeams.php [hunk 1]
```diff
@@ -36,7 +36,7 @@ public function delete($user)
     {
         DB::transaction(function () use ($user) {
             $this->deleteTeams($user);
-
+            $user->deleteProfilePhoto();
```
Comment (Chrissi2812) [inline stubs/app/Actions/Jetstream/DeleteUserWithTeams.php:39]: Could check before the deletion

```suggestion
            if (\Laravel\Jetstream\Features::managesProfilePhotos()) {
                $user->deleteProfilePhoto();
            }
```

File: stubs/app/Actions/Jetstream/DeleteUser.php [hunk 2]
```diff
@@ -14,6 +14,7 @@ class DeleteUser implements DeletesUsers
      */
     public function delete($user)
     {
+        $user->deleteProfilePhoto();
```
Comment (Chrissi2812) [inline stubs/app/Actions/Jetstream/DeleteUser.php:17]:
```suggestion
        if (\Laravel\Jetstream\Features::managesProfilePhotos()) {
                $user->deleteProfilePhoto();
       }
```

**Review by Chrissi2812 — CHANGES_REQUESTED** [review]
You could check if the Feature is enabled before, so no errors if it's not enabled

File: stubs/app/Actions/Jetstream/DeleteUser.php (hunk 2 above)
Comment (lostdesign) [reply stubs/app/Actions/Jetstream/DeleteUser.php]: Ohhh right, yeah just came across that feature check yesterday but didn't quite connect the dots here. Added it!

File: stubs/app/Actions/Jetstream/DeleteUser.php [hunk 3]
```diff
@@ -14,6 +15,9 @@ class DeleteUser implements DeletesUsers
      */
     public function delete($user)
     {
+        if (Features::managesProfilePhotos()) {
+            $user->deleteProfilePhoto();
+        }
         $user->delete();
```
Comment (claudiodekker) [inline stubs/app/Actions/Jetstream/DeleteUser.php:21]:
```suggestion
        }

        $user->delete();
```

File: stubs/app/Actions/Jetstream/DeleteUserWithTeams.php [hunk 4]
```diff
@@ -36,7 +37,9 @@ public function delete($user)
     {
         DB::transaction(function () use ($user) {
             $this->deleteTeams($user);
-
+            if (Features::managesProfilePhotos()) {
+                $user->deleteProfilePhoto();
+            }
             $user->delete();
```
Comment (claudiodekker) [inline stubs/app/Actions/Jetstream/DeleteUserWithTeams.php:43]:
```suggestion
            }

            $user->delete();
```

File: stubs/app/Actions/Jetstream/DeleteUserWithTeams.php [hunk 5]
```diff
@@ -36,6 +37,9 @@ public function delete($user)
     {
         DB::transaction(function () use ($user) {
             $this->deleteTeams($user);
+            if (Features::managesProfilePhotos()) {
```
Comment (claudiodekker) [inline stubs/app/Actions/Jetstream/DeleteUserWithTeams.php:40]:
```suggestion
            $this->deleteTeams($user);

            if (Features::managesProfilePhotos()) {
```

**Review by driesvints — APPROVED** [review]

**Review by claudiodekker — APPROVED** [review]

**Comment by taylorotwell** [comment]
Personally I would move that feature check into the `deleteProfilePhoto` method so you can just always call it from the stub.

**Comment by taylorotwell** [comment]
Looks like tests are failing...

**Comment by lostdesign** [comment]
@taylorotwell adjusted, guess that's the way you meant? The test now fails because the tests\Fixture\User.php  doesnt have the `HasProfilePicture` trait. Is that intended? Otherwise I'd add it so the test passes 👍 
https://github.com/laravel/jetstream/blob/e37dbff40a74d40580130f15d2cf401489580d2d/tests/Fixtures/User.php#L11

**Comment by taylorotwell** [comment]
You can add it to the fixture. Yes that looks fine. Thanks.

**Pull request merged**